ResearchKit/Common/ORKGraphView.m

```diff
@@ -0,0 +1,848 @@
+        delay += ORKGraphGrowAnimationDuration;
+    }
+    return delay;
+}
+
+- (void)animateLayer:(CAShapeLayer *)shapeLayer withAnimationType:(ORKGraphAnimationType)animationType {
+    [self animateLayer:shapeLayer withAnimationType:animationType toValue:1.0];
+}
+
+- (void)animateLayer:(CAShapeLayer *)shapeLayer withAnimationType:(ORKGraphAnimationType)animationType toValue:(CGFloat)toValue {
+    [self animateLayer:shapeLayer withAnimationType:animationType toValue:toValue startDelay:0.0];
+}
+
+- (void)animateLayer:(CAShapeLayer *)shapeLayer withAnimationType:(ORKGraphAnimationType)animationType startDelay:(CGFloat)delay {
+    [self animateLayer:shapeLayer withAnimationType:animationType toValue:1.0 startDelay:delay];
+}
+
+- (void)animateLayer:(CAShapeLayer *)shapeLayer withAnimationType:(ORKGraphAnimationType)animationType toValue:(CGFloat)toValue startDelay:(CGFloat)delay {
+    if (animationType == ORKGraphAnimationTypeFade) {
+        
+        CABasicAnimation *fadeAnimation = [CABasicAnimation animationWithKeyPath:@"opacity"];
+        fadeAnimation.beginTime = CACurrentMediaTime() + delay;
+        fadeAnimation.fromValue = @0;
+        fadeAnimation.toValue = @(toValue);
+        fadeAnimation.duration = ORKGraphFadeAnimationDuration;
+        fadeAnimation.timingFunction = [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionEaseOut];
+        fadeAnimation.fillMode = kCAFillModeForwards;
+        fadeAnimation.removedOnCompletion = NO;
+        [shapeLayer addAnimation:fadeAnimation forKey:FadeAnimationKey];
+        
+    } else if (animationType == ORKGraphAnimationTypeGrow) {
+        
+        CABasicAnimation *growAnimation = [CABasicAnimation animationWithKeyPath:@"strokeEnd"];
+        growAnimation.beginTime = CACurrentMediaTime() + delay;
+        growAnimation.fromValue = @0;
+        growAnimation.toValue = @(toValue);
+        growAnimation.duration = ORKGraphGrowAnimationDuration;
+        growAnimation.timingFunction = [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionEaseOut];
+        growAnimation.fillMode = kCAFillModeForwards;
+        growAnimation.removedOnCompletion = NO;
+        [shapeLayer addAnimation:growAnimation forKey:GrowAnimationKey];
+        
+    } else if (animationType == ORKGraphAnimationTypePop) {
+        
+        CABasicAnimation *popAnimation = [CABasicAnimation animationWithKeyPath:@"transform.scale"];
+        popAnimation.beginTime = CACurrentMediaTime() + delay;
+        popAnimation.fromValue = @0;
+        popAnimation.toValue = @(toValue);
+        popAnimation.duration = ORKGraphPopAnimationDuration;
+        popAnimation.timingFunction = [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionEaseOut];
+        popAnimation.fillMode = kCAFillModeForwards;
+        popAnimation.removedOnCompletion = NO;
+        [shapeLayer addAnimation:popAnimation forKey:GrowAnimationKey];
```

Comment:
Most of the configurations are duplicated among the if blocks except:
`CABasicAnimation *popAnimation = [CABasicAnimation animationWithKeyPath:@"transform.scale"]`
and
`popAnimation.duration = ORKGraphPopAnimationDuration;`

It would be nice to move other lines out of the `if(){...}else{...}` blocks.